test/common/http/conn_manager_impl_test.cc

```diff
@@ -833,6 +833,7 @@ TEST_F(HttpConnectionManagerImplTest, MultipleFilters) {
       .WillOnce(InvokeWithoutArgs([&]() -> Http::FilterHeadersStatus {
         EXPECT_EQ(route_config_provider_.route_config_->route_,
                   decoder_filter1->callbacks_->route());
+        EXPECT_EQ(ssl_connection_.get(), decoder_filter1->callbacks_->ssl());
```

Comment:
I added another check in the test above with ssl connection. So we are covering both ssl being nullptr and not null.